crawler.py

```diff
@@ -0,0 +1,54 @@
+from selenium import webdriver
+from bs4 import BeautifulSoup, SoupStrainer
+import unicodedata
+import re
+import traceback
+
+
+class Crawler:
+
+    def __init__(self, url):
+        self.url = url
+        self.queue = []
+        self.res = set()
+        self.email_regex = re.compile('([A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4})', re.IGNORECASE)
+
+    def connect(self):
+        driver = webdriver.Firefox()
+        driver.get("http://" + self.url)
+        html = driver.page_source
+        soup = BeautifulSoup(html)
+        for a in soup.find_all('a', href=True):
+            m = re.search(self.email_regex, a["href"])
```

Comment:
This will cause the crawler to walk through every page, regardless of whether or not the link is within the same domain as the one entered by the user.